truss/templates/control/control/endpoints.py

```diff
@@ -66,47 +60,60 @@ def proxy(path):
                 ):
                     error_msg = "It appears your model has stopped running. This often means' \
                         ' it crashed and may need a fix to get it running again."
-                    return Response(error_msg, 503)
+                    return JSONResponse(error_msg, 503)
                 raise exp
 
-    headers = [(name, value) for (name, value) in resp.raw.headers.items()]
-    response = Response(resp.content, resp.status_code, headers)
+    if _is_streaming_response(resp):
+        return StreamingResponse(
+            resp.aiter_bytes(), media_type="application/octet-stream"
+        )
+
+    response = Response(resp.content, resp.status_code, resp.headers)
     return response
 
 
-@control_app.route("/control/patch", methods=["POST"])
-def patch() -> Dict[str, str]:
-    current_app.logger.info("Patch request received.")
-    patch_request = request.get_json()
-    current_app.config["inference_server_controller"].apply_patch(patch_request)
-    current_app.logger.info("Patch applied successfully")
+control_app.add_route("/v1/{path:path}", proxy, ["GET", "POST"])
+
+
+@control_app.post("/control/patch")
+async def patch(request: Request) -> Dict[str, str]:
+    request.app.state.logger.info("Patch request received.")
+    patch_request = await request.json()
+    loop = asyncio.get_event_loop()
+    await loop.run_in_executor(
+        None,
+        lambda: request.app.state.inference_server_controller.apply_patch(
+            patch_request
+        ),
+    )
+    request.app.state.logger.info("Patch applied successfully")
     return {"msg": "Patch applied successfully"}
 
 
-@control_app.route("/control/truss_hash", methods=["GET"])
-def truss_hash() -> Dict[str, Any]:
-    t_hash = current_app.config["inference_server_controller"].truss_hash()
+@control_app.get("/control/truss_hash")
+async def truss_hash(request: Request) -> Dict[str, Any]:
+    t_hash = request.app.state.inference_server_controller.truss_hash()
     return {"result": t_hash}
 
 
-@control_app.route("/control/restart_inference_server", methods=["POST"])
-def restart_inference_server() -> Dict[str, str]:
-    current_app.config["inference_server_controller"].restart()
+@control_app.post("/control/restart_inference_server")
+async def restart_inference_server(request: Request) -> Dict[str, str]:
+    request.app.state.inference_server_controller.restart()
```

Comment:
does this also need to run in executor?

Reply:
Don't we want the restart operations to block the event loop so we should run them directly?